### ISSUE TITLE
submitting Form wiht PATCH method (isSubmitted()) not working

### ISSUE THREAD

**Issue by etshy** [symfony/symfony#40452]
**Symfony version(s) affected**: 5.2.3

**Description**

on a Form with a PATCH method, with the http_method_override options set to true, the form isn't marked as submitted.
The method property of the `Request` class is already set to POST and so it doesn't pass the first condition of the `handleRequest()` method.

**How to reproduce**

set the http_method_override to true and the forme method to PATCH.

**Possible Solution**

I didn't really look at it in details yet, but from the (dirty) debug I made, the method property of the `Request` class is already set to POST and so it doesn't pass the first condition of the `handleRequest()` method.

**Additional context**

`framework.yml`
```
framework:
  secret: '%env(APP_SECRET)%'
  #default_locale: en
  csrf_protection: ~
  http_method_override: true
```

`MyForm.php`
```
    public function buildForm(FormBuilderInterface $builder, array $options)
    {
        $builder->setMethod('PATCH');
        ...
```

`MyController`
```
 public function edit(Request $request)
    {
        $form = $this->createForm(MyForm::class, $class);
        $form->handleRequest($request);

        if ($form->isSubmitted()) { // Always false when using PATCH, works when using POST
```

**Comment by xabbuh** [comment]
Can you create a small example application that allows to reproduce your issue?

**Comment by etshy** [comment]
I'll do it as soon as I can, as I'm working on this on my free time I don't guarantee it will be ready soon.

Though i noticed something when creating the bug reproducer from symfony/skeleton (as doc says) thah could help identify the problem.
I noticed my project `framework.yml` file has an option `http_cache: true` while the fresh one doesn't have this option. -To be honest, I don't even remember adding this option so I didn't even consider it could be a problem)

When I comment the `http_cache` option, the PATCH method form submission works well so i guess it's this option that makes the form submission fail.
I don't know if it's intended behaviour but if it is, maybe it could be good  top indicates something about it in the doc ?

**Comment by jvasseur** [comment]
This is probably not related to the form component but instead some kind of conflict between the `http_method_override` and the `http_cache` option (if the `http_method_override` where working properly, the `method` attribute of your request would be `PATCH`).

**Comment by etshy** [comment]
Yeah jvasseur, seems to be somethings like that.

I made a quick reproducer repo with just a Form and a Controller.
https://github.com/etshy/symfony_issuer_40452

**Comment by xabbuh** [comment]
Status: Reviewed

**Comment by xabbuh** [comment]
I found an issue here. `getMethod()` of the `Request` class is called before `enableHttpMethodParameterOverride()` when the HTTP cache is enabled. I think we need to account for that and it will be fixed by #40497.

However, I think that this will not completely resolve the underlying issue. The result of the `getMethod()` call is used to calculate the cache key. In this example, that key will now be `POST` which is not correct and will lead to issues when there are two controllers for this request path doing different things depending on the request method.

**Comment by xabbuh** [comment]
I forgot to mention that the patch in #40497 was completely rewritten to call `enableHttpMethodParameterOverride()` inside `HttpCache::handle()` which fully solves this issue.
